### PR TITLE
Fix multiple-row manual moving.

### PR DESCRIPTION
**Pull request by jansiegel**
### Context
This should fix the bug from #6891. Additionally, fixed a bug with the `isParent` method and added test cases.

### How has this been tested?
Added test cases.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #6891

### CODE REVIEW

**Comment by jansiegel** [comment]
@budnix 
> It works but very slow. In my demo I'm using 25 rows and moving 5 rows between parents takes 2s, moving 20 rows takes 5s. Looking through profiler I've seen that there is a lot of `render` call. Unfortunately most of the time is spending on calculating column widths, IndexMapper bug? To investigate.

And how's performance on `develop` when moving a single row?

**Comment by budnix** [comment]
After removing the [render](https://github.com/handsontable/handsontable/pull/6935/files#diff-cc16ad5b7fc2e4c125a60c45ec844092R686) call and if the `setSourceDataAtCell` method wouldn't call `render` internally then the performance looks good.

**Comment by budnix** [comment]
> And how's performance on `develop` when moving a single row?

On `develop` moving the rows tasks ~200ms but for 10000 rows.

**Comment by jansiegel** [comment]
I guess we can make `render` in `setSourceDataAtCell` optional, WDYT @wojciechczerniak?

**Comment by budnix** [comment]
Originally I had the same [idea](https://github.com/handsontable/handsontable/pull/6930/commits/dea4150ce44f6e9904c467ac8919de60b197c9fc#diff-7eb52b366866677666470e019283c8eaR2399). But after thinking more about the fourth argument I came to the conclusion that maybe it's better to remove `render` call completely? In my PR it turned out that it's not necessary for me so I was removed the implementation.

**Comment by jansiegel** [comment]
> Originally I had the same [idea](https://github.com/handsontable/handsontable/pull/6930/commits/dea4150ce44f6e9904c467ac8919de60b197c9fc#diff-7eb52b366866677666470e019283c8eaR2399). But after thinking more about the fourth argument I came to the conclusion that maybe it's better to remove `render` call completely? In my PR it turned out that it's not necessary for me so I was removed the implementation.

Internally it probably works fine, but calling `setSourceDataAtCell` separately should probably update what's displayed in the table, just as `setDataAtCell` does, I guess?

**Comment by wojciechczerniak** [comment]
True, the idea was to have every public API method to consistently refresh the view.

I agree that in internal operations, or for some advance use cases the render operation is undesirable at least. I would consider it harmful for our performance and a bug.

API should be then extended with the possibility to batch operations or suspend/resume cache rebuild and render. Taking a little bit further the `executeBatchOperations` concept. Very similar idea to what we have in the HF:

```ts
public batch(callback): void {
  this.suspendRender();

  callback();

  this.resumeRender();
}
```

Or another use case, without the callback:

```ts
this.suspendRender();

this.setSourceDataAtCell( ... );

this.resumeRender();
```

But as we spoke with @budnix, it's not something we should do in a hurry, v8.1 maybe.

TL;DR;
I think that the best, for now, will be if we go with an additional argument (can we keep it private?). We will avoid a breaking change by adding/removing render on `setSourceDataAtCell` and we can refactor this later internally to `batch` operations.

**Comment by jansiegel** [comment]
> TL;DR;

I think that's reasonable. Private as in not describe it JSDoc or something more?

**Comment by wojciechczerniak** [comment]
Yes. IMO skipping the docs should be enough but it's up to the reviewer 

Look good. And maybe we can use the `silentMode` where @budnix used it before for some additional performance gain. There might be some other places that can use this optimization

**Comment by jansiegel** [comment]
> There might be some other places that can use this optimization

Probably, but I'd leave it for another task. Even adding `silentMode` to `setSourceDataAtCell` is a bit of a reach for a PR about moving rows.

**Comment by budnix** [comment]
@jansiegel After the changes the performance is much better 👏   However, while playing the demo a little bit I've found a bug. After moving all children from one parent node to another and after moving an empty parent and drop somewhere the code throws an exception.

nested-rows
Is this error should be handled by this PR?

**Comment by jansiegel** [comment]
> Is this error should be handled by this PR?

I think so, yes, I'll take a look at it.

**Comment by jansiegel** [comment]
https://github.com/handsontable/handsontable/pull/6935/commits/4e3990f6cef7fa8e72299273f983c6e1a1dbba87 blocks moving for the top-level rows in NestedRows (it works like that in `7.x`). 
I'm not exactly sure if it's supposed to work like that by design, so I did not add any tests - it might be misleading for the future changes.

I'm guessing the moving for parents and top-level rows should be eventually implemented, but it's not a thing for this PR.

File: src/plugins/nestedRows/test/nestedRows.e2e.js [hunk 1]
```diff
@@ -104,6 +104,19 @@ describe('NestedRows', () => {
       expect(getData()).toEqual(dataInOrder);
     });
 
+    it('should not move rows when they are on the highest level of nesting (don\'t have a parent)', () => {
```
Comment (budnix) [inline src/plugins/nestedRows/test/nestedRows.e2e.js:107]: I thought that this test checks if there is no error while parent moving https://github.com/handsontable/handsontable/pull/6935#issuecomment-637535697. However, after reverting the changes which block parents moving (and prevents errors) the test still passes.

I'd add a test for the specific case described in the above GitHub comment. There I removed all parent's children then I move the parent between top rows, then the error occurs.
